Add documentation page about the fetch service

ineesalmeida opened the pull request.

jugmac00 commented on reference/services/fetch-service.rst:
```diff
@@ -0,0 +1,149 @@
+
+To do so, a Launchpad admin is required to set the
+``use_fetch_service`` flag within a snap to ``true``, either in the API or in
+the UI by accessing the Admin area of a snap.
```
Admin -> admin

jugmac00 commented on reference/services/fetch-service.rst:
```diff
@@ -0,0 +1,149 @@
+Bug trackers
+------------
+* `Fetch service snap GitHub issues <https://github.com/canonical/fetch-service/issues>`_
+* `Fetch service charm GitHub issues <https://github.com/canonical/fetch-operator/issues>`_
```
Sometimes you capitalize snap and charm, sometimes not. I'd assume the capitalized version is the right one.

ineesalmeida replied:
I made it coherent now, I changed it to non-capitalized - I felt it made more sense

jugmac00 commented on reference/services/fetch-service.rst:
```diff
@@ -0,0 +1,149 @@
+---------
+Production
+~~~~~~~~~~
+Not set up.
```
Would IS be able to view the logs? If so, how?

ineesalmeida replied:
That's part of another ticket. I added details here, and added a comment in the ticket to update this section of the documentation

jugmac00 reviewed (changes requested):
Thanks a lot for the PR!

The fetch service repos are private, so the link checker fails. You need to add the links to https://github.com/canonical/launchpad-manual/blob/5969eac325484194150235caac48c6e186b55833/custom_conf.py#L112

To lower turnaround time, you can run the checks locally via make, e.g. `make linkcheck`.

ineesalmeida commented:
Thank you for the review, I've addressed all the comments and the checks are now passing.
I had some trouble getting the links to work though, I think something might be missing. To link between the build-farm and the fetch-service files, I used what is used in other documents:
```
See :doc:`Build Farm <../build-farm>` for more details
```
But when I run it locally, the link doesn't work.

When I do
```
See `Build Farm <../build_farm>`_ for more details
```
instead, the links works but the linkchecker fails. I need to further look into what's missing regarding the links.

simonepelosi commented:
> Thank you for the review, I've addressed all the comments and the checks are now passing. I had some trouble getting the links to work though, I think something might be missing. To link between the build-farm and the fetch-service files, I used what is used in other documents:
> 
> ```
> See :doc:`Build Farm <../build-farm>` for more details
> ```
> 
> But when I run it locally, the link doesn't work.
> 
> When I do
> 
> ```
> See `Build Farm <../build_farm>`_ for more details
> ```
> 
> instead, the links works but the linkchecker fails. I need to further look into what's missing regarding the links.

I pushed a small fix for that :) the path was wrong, the correct one is <build-farm> instead of <../build-farm>

ineesalmeida commented:
Thank you Simone! I fixed the remaining links, so it's fully ready now

jugmac00 reviewed (approved):
Thanks a lot! This looks great.
